src/test_gen/java_test_case_generator.cpp

```diff
@@ -0,0 +1,83 @@
+#include <iostream>
+#include <functional>
+
+#include <util/message.h>
+#include <cbmc/bmc.h>
+
+#include <java_bytecode/java_entry_point.h>
+#include <test_gen/java_test_source_factory.h>
+#include <test_gen/java_test_case_generator.h>
+
+namespace
+{
+inputst generate_inputs(const symbol_tablet &st, const goto_functionst &gf,
+    const goto_tracet &trace)
+{
+  std::map<const irep_idt, exprt> result;
+  for (const goto_trace_stept &step : trace.steps)
+    if (goto_trace_stept::ASSIGNMENT == step.type)
+      result.insert(std::make_pair(step.identifier, step.full_lhs_value));
+  return result;
+}
+
+const irep_idt &get_entry_function_id(const goto_functionst &gf)
```

Comment:
I found this function very hard to read, because there is no structuring, such as blank lines or the likes. It's doing something very simple, but I found that far from obvious.